aten/src/ATen/native/Convolution.cpp

```diff
@@ -186,10 +187,22 @@ auto ConvParams::is_depthwise(
 }
 
 static void check_shape_forward(const at::Tensor& input,
-                                      const at::Tensor& weight, const at::Tensor& bias,
-                                      const ConvParams& params) {
+                                const at::Tensor& weight, const at::Tensor& bias,
+                                const ConvParams& params, bool input_is_mkldnn) {
   int64_t k = input.ndimension();
   int64_t weight_dim = weight.ndimension();
+  std::vector<int64_t> weight_sizes(k);
```

Comment:
nit: maybe keep weight_sizes as ArrayRef and have a separate vector for sizes in mkldnn case and then point ArrayRef to that vector?

Reply:
Does ArrayRef extend the lifetime of a vector? (If not then we still can not avoid the creation of a vector?)

Reply:
ArrayRef doesn't extend the lifetime of the vector.